packages/aws-cdk-lib/region-info/lib/region-info.ts

```diff
@@ -70,6 +70,13 @@ export class RegionInfo {
     return Fact.find(this.name, FactName.CDK_METADATA_RESOURCE_AVAILABLE) === 'YES';
   }
 
+  /**
+   * Whether Amazon ECR Public is available in this region or not.
+   */
+  public get ecrPublicAvailable(): boolean {
+    return Fact.find(this.name, FactName.ECR_PUBLIC_AVAILABLE) === 'YES';
```

Comment:
So this would mean if a user doesn't specify a region in the stack, which is the common case, we now change the behavior and remove the policy and remove the ecr authorization - this might have a some implications no? We should default to the common case, and make the user "work" a little harder for the uncommon one.